pkg/yang/identity_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yang
+
+import "testing"
+
+type modIn struct {
+	name    string
+	content string
+}
+
+type modOut struct {
+	name       string
+	identities []*idOut
+}
+
+type idOut struct {
+	idName   string
+	baseName string
+}
+
+type testCase struct {
+	name string
+	in   []*modIn
+	out  *modOut
+	err  string
+}
+
+// A set of test modules to parse as part of the test case.
+var testCases = []*testCase{
+	&testCase{
+		in: []*modIn{
+			&modIn{
+				name: "idtest-one",
+				content: `
+module idtest-one {
+  namespace "urn:idone";
+  prefix "idone";
+
+  identity TEST_ID;
+}
+    `},
+		},
+		out: &modOut{
+			name: "idtest-one",
+			identities: []*idOut{
+				&idOut{
+					idName: "TEST_ID",
+				},
+			},
+		},
+		err: "tc1: could not resolve identities",
+	},
+	&testCase{
+		in: []*modIn{
+			&modIn{
+				name: "idtest-two",
+				content: `
+module idtest-two {
+  namespace "urn:idtwo";
+  prefix "idone";
+
+  identity TEST_ID;
+  identity TEST_ID_TWO;
+  identity TEST_CHILD {
+    base TEST_ID;
+  }
+}
+    `},
+		},
+		out: &modOut{
+			name: "idtest-two",
+			identities: []*idOut{
+				&idOut{
+					idName: "TEST_ID",
+				},
+				&idOut{
+					idName: "TEST_ID_TWO",
+				},
+				&idOut{
+					idName:   "TEST_CHILD",
+					baseName: "TEST_ID",
+				},
+			},
+		},
+		err: "could not resolve identities",
+	},
+}
+
+func TestIdentityExtract(t *testing.T) {
+	for _, testCase := range testCases {
+		ms := NewModules()
+		for _, mod := range testCase.in {
+			_ = ms.Parse(mod.content, mod.name)
+		}
+		parsedMod, err := ms.GetModule(testCase.out.name)
+
+		if err != nil {
+			t.Errorf("could not parse module: %s", testCase.out.name)
+			continue
+		}
+
+		identityNames := make(map[string]*Identity)
```

Comment:
I don't think  you need this map once you have out being a map:

for _, id := range parseMod.Identities {
    base, ok := tt.out[id.Name]
    delete(tt.out, id.Name)
    switch {
    case !ok:
        t.Errorf("%s: identity %q missing", tt.name, id.Name)
    case base != id.Base.Name:
        t.Errorf("%s: identity %q got base %q, want %q", tt.name, id.Name, id.Base.Name)
    }
}
for id := range tt.out {
    t.Errorf("%s: did not get expected identity %q", tt.name, id)
}